index.bs

```diff
@@ -1437,18 +1448,44 @@ object is discarded in the runtime, subsequent attempts to access it via the
 protocol will result in an error.
 
 <div algorithm>
-To get the <dfn>object id for an object</dfn> given |realm| and |object|:
+To get the <dfn>handle for an object</dfn> given |realm|, |ownership type| and
+|object|:
+
+1. If |ownership type| is equal "<code>none</code>", return <code>null</code>.
+
+1. Let |handle id| be a new, unique, string handle for |object|.
+
+1. Let |handle map| be |realm|'s [=handle object map=]
+
+1. Set |handle map|[|handle id|] to |object|.
+
+1. Return |handle id| as a result.
+
+</div>
+
+<div algorithm>
+To <dfn>set internal ids if needed</dfn> given |serialization internal map|,
+|remote value| and |object|:
+
+1. If the |serialization internal map| does not contain |object|, set
+   |serialization internal map|[|object|] to |remote value|.
 
-1. If the given |realm|'s [=object id map=] does not contain |object|, run the
-   following steps:
+1. Otherwise, run the following steps:
 
-  1. Let |object id| be a new, unique, string identifier for |object|. If |object| is an
-     [=/element=] this must be the [=web element reference=] for |object|; if it's a {{WindowProxy}}
-     object, this must be the [=browsing context id=] for |object|.
+    1. Let |previously serialized remote value| be
+       |serialization internal map|[|object|].
 
-  1. Set the value of |object| in |realm|'s [=object id map=] to |object id|.
+    1. If |previously serialized remote value| does not have a field
+       <code>internalId</code>, run the following steps:
 
-1. Return the result of getting the value for |object| in |realm|'s [=object id map=].
+      1. Let |internal id| be a unique across the <code>internalId</code> fields of
```

Comment:
Having `internalId` a number seems enough to prevent uses from trying to parse it or rely on their structure. Requiring them to be a consequent numbers doesn't seem to benefit users, but can complicate implementation. E.g. reserving the `internalId` and the moment of putting the object to the set vs using counter at the moment of duplication found produces different results.

Reply:
OK, if you feel like it's going to complicate your implementation we can make it arbitary.